src/redux/slices/keyring.ts

```diff
@@ -200,6 +200,58 @@ export const login = createAsyncThunk(
   }
 );
 
+export const importAccountFromPem = createAsyncThunk(
+  'keyring/importAccountFromPem',
+  async (
+    { pem, icon, name }: { name: string; icon: string; pem: string },
+    { rejectWithValue }
+  ) => {
+    try {
+      const instance = KeyRing.getInstance();
+      const response = await instance?.importAccountFromPem({
+        pem,
+        icon,
+        name,
+      });
+      return response;
+    } catch (e: any) {
+      // TODO: Add toast to handle error.
+      return rejectWithValue({ error: e.message });
```

Comment:
return the error directly, i've change that in the other actions
` return rejectWithValue(e.message);`